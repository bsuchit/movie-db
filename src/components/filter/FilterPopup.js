import React, { useEffect, useState } from "react";
import { List, Checkbox } from "semantic-ui-react";
import PropTypes from 'prop-types';
import _ from "lodash";

function FilterPopup({ data = [], selection = [], onSelected }){
    const [value, setValue] = useState();

    useEffect(() => {
        setValue(selection);
    }, [selection]);

    const handleOnChange = (e, data) => {
        onSelected(data.name, data.checked);
    };

    return <List>
        {
            _.map(data, ({ name, label }) => (
                <List.Item key={name} >
                    <Checkbox
                        name={name}
                        label={label}
                        checked={selection && name === value}
                        onChange={handleOnChange}
                        data-testid={name}
                    />
                </List.Item>
            ))
        }
  </List>;
}

FilterPopup.propTypes = {
    data: PropTypes.array,
    onSelected: PropTypes.func,
    selection: PropTypes.string
}

export default FilterPopup;