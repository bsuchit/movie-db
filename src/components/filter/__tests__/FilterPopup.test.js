import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import FilterPopup from "../FilterPopup";

const sortByData = [
    {
      name: 'episode_id',
      label: 'Episode'
    },
    {
      name: 'release_date',
      label: 'Release Date'
    }
  ];

describe('Given FilterPopup', () => {
    const mockOnSelected = jest.fn();

    it('should render correctly with given props', () => {
        const { container } = render(<FilterPopup data={sortByData} selection={null} onSelected={mockOnSelected} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should call onSelected when checkbox is clicked', () => {
        render(<FilterPopup data={sortByData} selection={null} onSelected={mockOnSelected} />);
        const checkbox = screen.getByTestId('episode_id');
        userEvent.click(checkbox.childNodes[0]);
        expect(checkbox.childNodes[0]).toBeChecked();
        expect(mockOnSelected).toHaveBeenCalledWith('episode_id', true);
    });
});