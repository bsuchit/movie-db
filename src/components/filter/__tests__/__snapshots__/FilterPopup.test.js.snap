// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given FilterPopup should render correctly with given props 1`] = `
<div
  class="ui list"
  role="list"
>
  <div
    class="item"
    role="listitem"
  >
    <div
      class="ui checkbox"
      data-testid="episode_id"
    >
      <input
        class="hidden"
        name="episode_id"
        readonly=""
        tabindex="0"
        type="checkbox"
        value=""
      />
      <label>
        Episode
      </label>
    </div>
  </div>
  <div
    class="item"
    role="listitem"
  >
    <div
      class="ui checkbox"
      data-testid="release_date"
    >
      <input
        class="hidden"
        name="release_date"
        readonly=""
        tabindex="0"
        type="checkbox"
        value=""
      />
      <label>
        Release Date
      </label>
    </div>
  </div>
</div>
`;
